algviz/picture/connection_map.py

```diff
@@ -0,0 +1,136 @@
+import svgutils.transform as svgutils
+import os
+
+from algviz.parser import structures
+from .anchor import Anchor
+from .svg_engine import SVGEngine
+
+"""Tracks connection points and connection pictures to be composed into a single picture"""
+
+pointers_temp_file = "temp_pointers.svg"
+
+class ConnectionMap:
+    class Connection:
+        def __init__(self, start_point, connect_pic, anchor):
+            self.start_point = start_point
+            self.connect_pic = connect_pic
+            if not connect_pic.is_drawn:
+                connect_pic.draw()
```

Comment:
It's counterintuitive for pictures to `draw` themselves.  http://www.mcescher.com/gallery/back-in-holland/drawing-hands/